src/consensus/encode.rs

```diff
@@ -545,21 +545,35 @@ impl Encodable for [u16; 8] {
     }
 }
 
+#[inline]
+fn encode_vec<S: io::Write, T: Encodable>(mut s: S, vec: &Vec<T>) -> Result<usize, io::Error> {
```

Comment:
Dumb question, but why even a `&Vec<T>` instead of `&[T]`?

Reply:
I would usually but I wasn't sure if we wanted to stick with only vec being encodable or expand it to include slices.

Reply:
Taking `&[T]` is strictly better AFAIK, and taking `&Vec<T>` is like taking `&String` are 99.9% non-idiomatic. I thought there were `cargo clippy` error for it, but I might be wrong.